db/database.go

```diff
@@ -162,6 +162,13 @@ func (d *PgxDatabase) Transact(fn TxFunc, opts ...TxOpt) (interface{}, error) {
 	}
 	var attempt int
 	var ret interface{}
+	var tx pgx.Tx
+	defer func() {
+		if p := recover(); p != nil && tx != nil {
+			_ = tx.Rollback(options.ctx)
+			panic(p)
```

Comment:
Does this lose the panic stacktrace (or is it stored on `p`?)?  If so, can we keep it somehow?

Reply:
Verified that the stack trace captures the original panic stack